agenta-web/src/components/Playground/ViewNavigation.tsx

```diff
@@ -66,13 +78,51 @@ const ViewNavigation: React.FC<Props> = ({
             variantContainerPath()
         }
 
+        const restartContainerHandler = async () => {
+            // Set restarting to true
+            setRestarting(true)
+
+            // Set payload to send to backend
+            const data: RestartVariantDocker = {
+                app_name: appName.toLowerCase(),
+                variant_name: variant.variantName,
+            }
+            try {
+                const response = await restartAppVariantContainer(data)
+                if (response.status === 200) {
+                    showNotification({
+                        type: "success",
+                        message: "App Container",
+                        description: `${response.data.message}`,
+                        duration: 5,
+                        key: response.status,
+                    })
+
+                    // Set restarting to false
+                    setRestarting(false)
+                    setTimeout(() => {
```

Comment:
Can we try using the short polling method to detect when the container is up and running as we do when starting the container from the apps page?

Reply:
Yes, yes. Could you please point to me where this was implemented in the codebase? Thank you.

Reply:
So you can find a function named `waitForAppToStart` in the api.ts file, that implements short polling. One you restart the container, you can do `await waitForAppToStart(appName)` and then try reloading the page.

Reply:
Alright, alright. I'll check it out now. Thank you.